docs/develop/writingtests.rst

```diff
@@ -0,0 +1,79 @@
+.. _writingtests:
+
+Writing tests
+=============
+
+Tests in MicroPython are written in the path ``py/tests``:
+
+.. code-block:: bash
+   
+   .
+    ├── basics
+    ├── cmdline
+    ├── cpydiff
+    ├── esp32
+    ├── extmod
+    ├── feature_check
+    ├── float
+    ├── import
+    ├── inlineasm
+    ├── internal_bench
+    ├── io
+    ├── jni
+    ├── micropython
+    ├── misc
+    ├── multi_bluetooth
+    ├── multi_net
+    ├── net_hosted
+    ├── net_inet
+    ├── perf_bench
+    ├── pyb
+    ├── pybnative
+    ├── qemu-arm
+    ├── README
+    ├── run-internalbench.py
+    ├── run-multitests.py
+    ├── run-natmodtests.py
+    ├── run-perfbench.py
+    ├── run-tests
+    ├── run-tests-exp.py
+    ├── run-tests-exp.sh
+    ├── stress
+    ├── thread
+    ├── unicode
+    ├── unix
+    └── wipy
```

Comment:
I'm not sure there's much use listing the entire tests directory.... maybe just list the key dirs (basics, extmod, micropython, float) as well as the run-tests runner script?